lib/puppet/pops/evaluator/compare_operator.rb

```diff
@@ -222,7 +222,7 @@ def match_Hash(hash, left, scope)
 
   def match_Symbol(symbol, left, scope)
     return true if symbol == :default
-    equals(left, default, scope)
+    match(left, default, scope)
```

Comment:
Almost agree with @hlindberg , but it should say `equals(symbol, left)`, i.e. no `scope` parameter.